knowledge_base/consistent-index-settings.json

```diff
@@ -0,0 +1,4 @@
+{
+  "id": "consistent-index-settings",
+  "description": "An index should have the same core index settings and associated data values before and after a migration or upgrade."
```

Comment:
Is this actually true, or are we just asserting it as true in a "come at me, bro" sense and seeing if it sticks?  Happy either way, and it will be fascinating to see how it plays out.

The concern comes in what, specifically, we're checking for as I can see these settings changing (either the options available or how they're stored) as versions change.  This is probably a case of "unknown unknowns" since we haven't delved into how different versions express this, so maybe it does make sense not to tie this expectation to specific versions and see how it plays out...

Curious on your thoughts @gregschohn

Reply:
This is definitely a leading expectation as I am not aware of how this is handled in all versions. This is specifically targeted at index settings which I have at least some confidence on a couple values here as they are included in the bwc tests which makes no distinction really between versions. My plan was identifying this expectation as the core list of expectations that remains consistent and as we find specific settings that change or get added we could create specific expectations for that setting if that makes sense. But this is definitely a good one to think about as we will have similar issues

Reply:
I've got some thoughts below - but I think that we should add some informative indicators to show the confidence in our assertions - from - maybe/hopeful thinking/I'd like to we've tested this elsewhere a million times.  

Another axis for informative purposes might be sensitivity.  That would probably be baked into the description.  Those additional axes/fields can grow later & we don't need to discuss that here.